CMakeLists.txt

```diff
@@ -231,12 +232,21 @@ endif ()
 # library.
 add_definitions(-DBOOST_TEST_DYN_LINK)
 
-# Detect OpenMP support in a compiler. If the compiler supports OpenMP, the
-# flags to compile with OpenMP are returned and added.
+# Detect OpenMP support in a compiler. If the compiler supports OpenMP, flags 
+# to compile with OpenMP are returned and added and the HAS_OPENMP is set to 1. 
+# This way we can skip calls to functions defined in omp.h with code like:
+# if (HAS_OPENMP == 1) { openMP code here }
+# If OpenMP is found, define HAS_OPENMP to be 1. Otherwise define it to be 0.
 find_package(OpenMP)
 if (OPENMP_FOUND)
+  add_definitions(-DHAS_OPENMP)
+  set(HAS_OPENMP "1")
```

Comment:
I am defining it here to use it in `lsh_search_impl.hpp`, where I check if HAS_OPENMP == 1 or HAS_OPENMP == 0. I'm not sure this is the correct usage for `add_definitions()` - I just saw it worked and went with it.

If I remove the num_threads() directive from the pragmas, then I guess this whole thing is skippable. num_threads() was added because I was planning to do the whole hybrid parallelization scheme, but now that's out, I can simplify the code a bit.